src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/CSharpFormattingPassBase.cs

```diff
@@ -347,23 +333,58 @@ private static bool ShouldFormatCore(FormattingContext context, TextSpan mapping
             return true;
         }
 
-        if (IsRazorComment() ||
-            IsInBoundComponentAttributeName() ||
-            IsInHtmlAttributeValue() ||
-            IsInDirectiveWithNoKind() ||
-            IsInSingleLineDirective() ||
-            IsImplicitExpression() ||
-            IsInSectionDirectiveCloseBrace() ||
-            (!allowImplicitStatements && IsImplicitStatementStart()))
+        if (IsInsideRazorComment())
```

Comment:
Just spread this out to aid debugging as it was impossible to tell which one of these methods was the thing that returned early. One of these "Should have done it years ago" changes.